### PR TITLE
Fix source generator crash on null DataRow arguments

### PR DESCRIPTION
**Pull request by Sergio0694**
## Summary
Fixes the NullReferenceException from MSTestReflectionMetadataGenerator when an attribute argument is a null array, including [DataRow(null)]. Roslyn exposes a default ImmutableArray for these arguments, so enumerating TypedConstant.Values crashes the generator.

- Short-circuit null arrays without enumerating their values, while checking accessibility of array element types used in generated casts and literals.
- Match the unit-test DataRowAttribute stub to the real constructor overloads so null binds to the array overload instead of masking the bug.
- Cover positional and named null arrays, nested arrays, inaccessible element types, and materialization as exactly one null argument.
- Extend managed and Native AOT acceptance tests to verify the null row executes exactly once.
- Make Windows crash-dump setup create the WER LocalDumps registry key when the hosted agent image does not provide it, allowing the build to proceed.

Minimal reproducer (no linked GitHub issue was provided):
```csharp
[TestMethod]
[DataRow(null)]
public void AcceptsNull(string? value) => Assert.IsNull(value);
```

## Validation
- All 170 source-generator unit tests passed.
- Managed source-generation acceptance test passed on net10.0.
- Native AOT acceptance test passed on net10.0 / win-x64, including publishing with both MSBuild and compiler warnings treated as errors. No IL warning suppressions were added.
- Re-ran `MSTest.SourceGeneration.UnitTests` after review: build and tests succeeded with 0 warnings and 0 errors.
- Confirmed the Windows CI failure occurs before compilation on main and unrelated PRs when the LocalDumps registry key is absent; the setup script now creates it idempotently.

### CODE REVIEW

File: src/Analyzers/MSTest.SourceGeneration/Generators/AttributeMaterializationHelper.cs [hunk 1]
```diff
@@ -246,7 +246,9 @@ private static bool IsAttributeMaterializable(AttributeData attribute, INamedTyp
     private static bool AreArgumentTypesReferenceable(TypedConstant constant, IAssemblySymbol consumingAssembly)
         => constant.Kind switch
         {
-            TypedConstantKind.Array => constant.Values.All(element => AreArgumentTypesReferenceable(element, consumingAssembly)),
+            // Null arrays have default Values, but their typed null casts still name the element type.
```
[the remark this answers is not in the document]
Comment (Sergio0694) [reply src/Analyzers/MSTest.SourceGeneration/Generators/AttributeMaterializationHelper.cs:249]: 🤖 Restored the UTF-8 BOM in f9406bd to match `.editorconfig`. Verified that the file starts with `EF BB BF`.

File: test/UnitTests/MSTest.SourceGeneration.UnitTests/MSTestReflectionMetadataGeneratorTests.cs [hunk 2]
```diff
@@ -2610,21 +2612,146 @@ namespace Sample
...
+        registry.Should().NotContain("AreAttributesComplete = false");
+    }
+
+    [TestMethod]
+    [DataRow("new Hidden[0]")]
+    [DataRow("new[] { Hidden.First }")]
+    [DataRow("new object[] { new Hidden[0] }")]
+    [DataRow("Value = new Hidden[0]")]
+    [DataRow("Value = new[] { Hidden.First }")]
+    [DataRow("Value = new object[] { new Hidden[0] }")]
```
[the remark this answers is not in the document]
Comment (Sergio0694) [reply test/UnitTests/MSTest.SourceGeneration.UnitTests/MSTestReflectionMetadataGeneratorTests.cs]: 🤖 Added all four requested positional/named, top-level/nested null-array cases in f9406bd. Roslyn types these boxed nulls as `object`, so the tests assert safe `(object)null!` materialization, complete attribute metadata, and no private `Hidden[]` reference. The existing non-null inaccessible-array cases still assert omission and incomplete metadata.

**Review by Evangelink — COMMENTED** [review]
> [!NOTE]
> 🤖 **Automated review by GitHub Copilot.** To request a follow-up action, reply by tagging `@copilot` directly.

### Review result: no actionable findings

The null-array handling is correctly ordered: `TypedConstant.Values` is no longer enumerated for null arrays, while the emitted cast's element type is still checked for accessibility. The updated `DataRowAttribute` stub matches the shipped overload set, and the tests verify generated-source compilation, exact one-null-argument materialization, nested arrays, named arguments, and inaccessible element types. Managed and Native AOT acceptance coverage also verifies that the row executes exactly once.

I independently built and ran `MSTest.SourceGeneration.UnitTests` at `f9406bd0b340030e5da2d3b1a7f62a1fc1ec7e66`; the project completed with 0 warnings and 0 errors. The current Windows CI failures occur in the pre-build **Enable local dumps** setup step because the `LocalDumps` registry path is absent; they are not caused by this diff.

<details>
<summary>Review coverage</summary>

- Scope and description alignment: clean; all four changed files support the stated fix.
- Algorithm and Roslyn typed-constant semantics: null, typed null, boxed null, nested array, empty array, and named-argument paths checked.
- Generated-code compatibility: emitted casts/literals, constructor overload binding, and inaccessible-type fallback checked.
- Public API and package compatibility: no public API or package metadata changes.
- Cross-TFM/AOT behavior: generator remains `netstandard2.0`; managed and Native AOT consumer paths covered.
- Test quality: assertions exercise compilation, emitted text, runtime materialization, discovery count, filter count, and exit code; no parallel-state or timing hazards introduced.
- Security, concurrency, resources, localization, CLI, MSBuild, and dependencies: no changed boundary or applicable regression.
- Existing review threads: both prior findings are resolved and were not duplicated.

</details>

**Review by Evangelink — APPROVED** [review]
Reviewed current head `497b4313b`. No actionable findings remain.

The generator now avoids enumerating Roslyn's default `TypedConstant.Values` for null arrays while still validating every type that generated casts and array literals reference. The `DataRowAttribute` test stub matches the shipped overload set, and the tests cover positional, named, nested, boxed, inaccessible-element, managed, and Native AOT paths with runtime materialization assertions. The Windows dump setup is idempotent, and the rerun passed on Windows Debug/Release, Linux, macOS, and application-model acceptance.

<details>
<summary>Review coverage</summary>

- Scope and PR description alignment
- Roslyn typed-constant and overload-resolution semantics
- Generated-source compilation and inaccessible-type fallback
- Managed and Native AOT execution behavior
- Test assertions, isolation, and mutation-sensitive edge cases
- Public API, compatibility, localization, CLI, dependencies, security, and concurrency
- Windows pipeline hardening and complete CI results
- Existing review threads and current branch state

</details>
